linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -3263,7 +3267,10 @@ private boolean advance() {
         TKey leftKey2 = outerKeySelector.apply(left);
         int c = leftKey.compareTo(leftKey2);
         if (c != 0) {
-          assert c < 0 : "not sorted";
+          if (c < 0) {
```

Comment:
The condition should be `c > 0`.
Original assertion was `c<0==true`, and assertion is violated in `c>0` case.

Reply:
oh, yes, sorry, trivial error !!!
So it should be "c >= 0" strictly speaking

Reply:
`c==0` is acceptable.